cranelift/codegen/src/ir/function.rs

```diff
@@ -277,14 +277,12 @@ impl FunctionStencil {
     /// Rewrite the branch destination to `new_dest` if the destination matches `old_dest`.
     /// Does nothing if called with a non-jump or non-branch instruction.
     pub fn rewrite_branch_destination(&mut self, inst: Inst, old_dest: Block, new_dest: Block) {
-        match self.dfg.insts[inst] {
+        match &mut self.dfg.insts[inst] {
```

Comment:
This change is where #5730 really shines, as we can avoid matching the `InstructionData` twice.

Reply:
Not that we couldn't have done that before, but removing `BranchInfo` ended up forcing the refactoring.